ifacemonitor/ifacemonitor_test.go

```diff
@@ -39,8 +40,9 @@ type netlinkTest struct {
 	addrUpdates    chan netlink.AddrUpdate
 	userSubscribed chan int
 
-	nextIndex int
-	links     map[string]linkModel
+	nextIndex  int
+	links      map[string]linkModel
+	linksMutex *sync.Mutex
```

Comment:
I believe it's idiomatic to use `sync.Mutex` instead of `*sync.Mutex`, then you can take advantage of the fact that a `zero` mutex is ready for use.